homeassistant/components/sensor/dominionenergy.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Washington Dominion Energy."""
+
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and getting the username and password."""
+    add_devices([DominionEnergySensor(config['username'], config['password'])])
+
+
+class DominionEnergySensor(Entity):
+    """Washington Dominion Energy Sensor will check the bill on daily basis."""
+
+    from datetime import timedelta
+    HOURS_TO_UPDATE = timedelta(hours=24)
+    CURRENT_BILL_SELECTOR = str("#homepageContent > div:nth-child(3) >" +
+                                " div:nth-child(2) > p > span")
+
+    def __init__(self, username, password):
+        """Initialize the sensor."""
+        self._state = None
+        self._password = password
+        self._username = username
+
+    @property
+    def name(self):
+        """Name of the Sensor: DominionEnergy Sensor."""
+        return "Dominion Energy"
+
+    @property
+    def state(self):
+        """Returning the State of DominionEnergy Sensor."""
+        return self._state
+
+    @Throttle(HOURS_TO_UPDATE)
+    def update(self):
+        """Using Selenium to access Dominion website and fetch data."""
+        from selenium import webdriver
```

Comment:
All requirements needs to be pinned -> https://home-assistant.io/developers/code_review_platform/#1-requirements

Also, you should consider to use requests/aiohttp to retrieve the data. Those modules are already available.

Reply:
they don't provide API that why I use selenium ( same case as the USPS component)

Reply:
Can you check if it's possible to use `requests`? Perhaps with something like this:

```python
with requests.Session() as session:
    post = session.post('https://www.dominionenergy.com/sign-in', data={"user": self._username, "password": self._password})
    response = session.get(REQUEST-URL, timeout=5)
    print(response.text)
```

The problem is PhantomJS. You can't use the distribution's package management tool to install it and in a container it will not be available.